### DISCUSSION TITLE
Confuse with the dpgen init_surf procedure

### DISCUSSION THREAD

**Discussion by ysu-007** [deepmodeling/dpgen#927] [Q&A]
**Summary**

**DPGEN Version and Platform**

**Job submission and computing cluster configuration**

**Details**

In the dpgen manual, there are only two steps for the dpgen init_surf:
_**Basically init_surf can be devided into two parts , denoted as stages in PARAM:

1. Build specific surface in folder 00.place_ele
2. Pertub and scale in folder 01.scale_pert_**_

For the first step, the software generates some initial surfaces with a vacuum and calculates their energy with DFT method.  Then the second step will give perturbation and scale to the initial structure generated in the first step, so are there any DFT calculations for the new structures generated in the second step? I don't find any DFT calculation yet in my test.  For the dpgen init_bulk,  I know there will be AIMD calculations for the structures generated in the second step which is step 3. However, there is no step 3 in init_surf.

**Comment by ysu-007** [comment]
What I see from the code is that the init_surf generates a lot of structures but doesn't calculate the DFT energy. Does the dpgen do the DFT calculation in another process?

**Reply by HuangJiameng** [reply]
Generally, `init_surf` does not run AIMD but only generates a lot of configurations. Compared with `init_bulk`, which runs DFT calculations twice, init_surf does once. Usually, we do `init_bulk`, run many rounds of DP-GEN iterations, collect enough data for the bulk system, and do `init_surf` after that. At this point, the lattice constant has been determined, and the lattice constant required for the initial configuration of `init_surf` can be used directly. These configurations made by `init_surf` are prepared for `01.model_devi`. Candidates will do DFT calculation in `02.fp`. (Thanks to @Carrotkingdom for help!) By the way, I think this topic is more proper to be start as a discussion... see https://github.com/deepmodeling/dpgen/discussions/new

**Reply by phyoung123** [reply]
Dear Huang @HuangJiameng , I am confused of the DFT calculations between the `init_bulk` and `Labeling`. In the `init_bulk` step, we can prepare the initio data set for different temperatures with MD runs, such for 3000K, 5000K, 8000K, but when I set different temperatures in `Exploration` step, How can I treat these temperatures in `Labeling` step？Looking forward to your helpful reply! Thank you.

**Reply by ysu-007** [reply]
Can dpgen automatically add the structure generated by init_surf to 01.model_devi of dpgen run, or do I need to copy it manually?

**Reply by HuangJiameng** [reply]
> Can dpgen automatically add the structure generated by init_surf to 01.model_devi of dpgen run, or do I need to copy it manually?

You have to do `dpgen run` after `dpgen init_surf`. Just make sure that everything you need is under the work path.

**Reply by HuangJiameng** [reply]
> Dear Huang @HuangJiameng , I am confused of the DFT calculations between the `init_bulk` and `Labeling`. In the `init_bulk` step, we can prepare the initio data set for different temperatures with MD runs, such for 3000K, 5000K, 8000K, but when I set different temperatures in `Exploration` step, How can I treat these temperatures in `Labeling` step？Looking forward to your helpful reply! Thank you.

You can compare these examples. [INCAR_methane.rlx](https://github.com/deepmodeling/dpgen/blob/check_failed_tasks/examples/init/INCAR_methane.rlx)(for stage 1 of init_bulk), [INCAR_methane.md](https://github.com/deepmodeling/dpgen/blob/check_failed_tasks/examples/init/INCAR_methane.md)(for stage 3 of init_bulk) and [INCAR_methane](https://github.com/deepmodeling/dpgen/blob/master/examples/run/dp1.x-lammps-vasp/CH4/INCAR_methane)(for scf calculation in labeling(02.fp) step)

**Reply by HuangJiameng** [reply]
The candidates calculated in `Labeling`(`02.fp`) step is picked from the result of `Exploration`(`01.model_devi`) step. They have different temperature and press.

**Reply by phyoung123** [reply]
Dear Huang @HuangJiameng , I really appreciate you for your helpful explanation.  For my understanding from your words, I just need to prepare the `fparam.raw`, which consists of the temperature of each frame, and set the temperature values in `Exploration` step. The `SIGMA` setting in `INCAR` for `02.fp` is doesn't matter, right? btw, the unit of temperature values in `fparam.raw` should be `K` or `eV`? Please excuse my rambling

**Reply by ysu-007** [reply]
> > Can dpgen automatically add the structure generated by init_surf to 01.model_devi of dpgen run, or do I need to copy it manually?
> 
> You have to do `dpgen run` after `dpgen init_surf`. Just make sure that everything you need is under the work path.

Thanks a lot.

**Reply by HuangJiameng** [reply]
> Dear Huang @HuangJiameng , I really appreciate you for your helpful explanation. For my understanding from your words, I just need to prepare the `fparam.raw`, which consists of the temperature of each frame, and set the temperature values in `Exploration` step. The `SIGMA` setting in `INCAR` for `02.fp` is doesn't matter, right? btw, the unit of temperature values in `fparam.raw` should be `K` or `eV`? Please excuse my rambling

1. The parameters you set in `INCAR` will be applied in DFT calculation, just like run VASP scf calculation manually. see https://github.com/deepmodeling/dpgen/blob/8dea29ef125f66be9641afe5ac4970433a9c9ce1/dpgen/generator/run.py#L2165-L2179
2. You can refer to [deepmd-kit/examples/fparam](github.com/deepmodeling/deepmd-kit/tree/master/examples/fparam), the "fitting_net" of "default_training_param" in [dpgen/examples/run/dp1.x-lammps-vasp-et](https://github.com/deepmodeling/dpgen/tree/master/examples/run/dp1.x-lammps-vasp-et) and a Chinese blog on [how to use `fparam`](https://zhuanlan.zhihu.com/p/349458538)

**Reply by phyoung123** [reply]
OK, I got it! Thank you so much~ @HuangJiameng

**Reply by lnnbig** [reply]
> Dear Huang @HuangJiameng , I really appreciate you for your helpful explanation. For my understanding from your words, I just need to prepare the `fparam.raw`, which consists of the temperature of each frame, and set the temperature values in `Exploration` step. The `SIGMA` setting in `INCAR` for `02.fp` is doesn't matter, right? btw, the unit of temperature values in `fparam.raw` should be `K` or `eV`? Please excuse my rambling

**I have also raised my question in DeePMD. Please disregard the message below if you come across it there.**

May I inquire if you have resolved the fparam issue? I am also confused by it.

Firstly, according to the file found at this link: https://github.com/deepmodeling/deepmd-kit/blob/master/examples/fparam/lmp/in.lammps, the line "pair_style deepmd frozen_model.pb fparam 0.25852028" suggests that the unit for fparam is clearly expressed in electron volts (eV). However, when I generated the Lammps file using DP-GEN, the unit indicated was in Kelvin (K).

Furthermore, there seems to be a significant problem with the value of max_devi_f in the model_devi.out file, which is remarkably small (1e-14). This suggests that there may be bugs present in both DPGEN (Version: 0.9.2) and DeePMD (v2.0.1).

**Reply by phyoung123** [reply]
No 1.  The electronic temperature `eV` in lammps input file do not conflict with fparam, in which, However, actually is `K`, you can check it in DPGEN auto-generated file after `Labeling` step.
No 2. I don't really know, maybe you can update your DPGEN version.

**Reply by lnnbig** [reply]
> No 1. The electronic temperature `eV` in lammps input file do not conflict with fparam, in which, However, actually is `K`, you can check it in DPGEN auto-generated file after `Labeling` step. No 2. I don't really know, maybe you can update your DPGEN version.

I double checked the fparam,npy in the example at this link: https://github.com/deepmodeling/deepmd-kit/tree/master/examples/fparam/data/e3000_i2000/set.000, which is showed below:

>>> np.load('fparam.npy')
array([0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028,
       0.25852028, 0.25852028, 0.25852028, 0.25852028, 0.25852028],
      dtype=float32)

which is in eV, not in K.

**Reply by lnnbig** [reply]
> No 1. The electronic temperature `eV` in lammps input file do not conflict with fparam, in which, However, actually is `K`, you can check it in DPGEN auto-generated file after `Labeling` step. No 2. I don't really know, maybe you can update your DPGEN version.

Thank you very much for your reply. May I please ask which version of DP-GEN are you using?

**Reply by AnguseZhang** [reply]
There're several definitions to be clarifed. 
1. There's no need to care about the unit of`fparam`, as long as the unit is consistent in all data you trained. So its unit can either be eV or K, or other units according to its physical meaning. Since it only indicates an additional input parameter of neuron network, apart of coordinates.
3. The energy unit in deep potential is eV, and force unit is eV/A.  
4. In lammps, the temperature unit is K, so what you see in LAMMPS may be the unit of temperature.
